contracts/USDCFlow.cdc

```diff
@@ -18,159 +20,207 @@ import "FiatToken"
+                    ftVaultData: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
+                )
+            case Type<FungibleTokenMetadataViews.FTDisplay>():
+                // Potential TODO: Replace with USDC http file and media type?
+                let media = MetadataViews.Media(
+                        file: MetadataViews.HTTPFile(
+                        url: "https://assets.website-files.com/5f6294c0c7a8cdd643b1c820/5f6294c0c7a8cda55cb1c936_Flow_Wordmark.svg"
+                    ),
+                    mediaType: "image/svg+xml"
+                )
+                let medias = MetadataViews.Medias([media])
+                return FungibleTokenMetadataViews.FTDisplay(
+                    name: "USDC (Flow)",
+                    symbol: "USDCf",
+                    description: "This fungible token representation of USDC is bridged from Flow EVM.",
+                    externalURL: MetadataViews.ExternalURL("https://www.circle.com/en/usdc"),
+                    logos: medias,
+                    socials: {
+                        "x": MetadataViews.ExternalURL("https://x.com/circle")
+                    }
+                )
+            case Type<FungibleTokenMetadataViews.FTVaultData>():
+                return FungibleTokenMetadataViews.FTVaultData(
+                    storagePath: USDCFlow.VaultStoragePath,
+                    receiverPath: USDCFlow.ReceiverPublicPath,
+                    metadataPath: USDCFlow.VaultPublicPath,
+                    receiverLinkedType: Type<&USDCFlow.Vault>(),
+                    metadataLinkedType: Type<&USDCFlow.Vault>(),
+                    createEmptyVaultFunction: (fun(): @{FungibleToken.Vault} {
+                        return <-USDCFlow.createEmptyVault(vaultType: Type<@USDCFlow.Vault>())
+                    })
+                )
+            case Type<FungibleTokenMetadataViews.TotalSupply>():
+                return FungibleTokenMetadataViews.TotalSupply(
+                    totalSupply: USDCFlow.totalSupply
+                )
+        }
+        return nil
+    }
+
+    access(all) resource Vault: FungibleToken.Vault {
 
         /// The total balance of this vault
-        pub var balance: UFix64
+        access(all) var balance: UFix64
 
         /// Initialize the balance at resource creation time
         init(balance: UFix64) {
             self.balance = balance
         }
 
+        /// Called when a fungible token is burned via the `Burner.burn()` method
+        /// The total supply will only reflect the supply in the Cadence version
+        /// of the FiatToken smart contract
+        access(contract) fun burnCallback() {
+            if self.balance > 0.0 {
+                USDCFlow.totalSupply = USDCFlow.totalSupply - self.balance
```

Comment:
same comment as below about `.burn`

Also, do we want to emit `Burned` here when `balance > 0.0`?

Reply:
both good suggestions! I added them to `burnCallback`